### ISSUE TITLE
parser: An index signature must have exactly one parameter.(1096)

### ISSUE THREAD

**Issue by Dunqing** [oxc-project/oxc#11052]
The following code typescript reports an error as `An index signature must have exactly one parameter.`

```ts
interface A {
  [A: number, B: number]: number;
}
```

Whereas Oxc reports an error as `Expected ']' but found ','`.

Current `TSIndexSignature`'s parameters is a `Vec`, which means it can store more than one parameter, but it wouldn't parse when it has more than one parameter, so that Vec here is unnecessary; we should change it to `TSIndexSignatureName` instead.

https://github.com/oxc-project/oxc/blob/c1e6c9bfa2ff505dcc87c98495c7fb78185a217d/crates/oxc_ast/src/ast/ts.rs#L1007-L1013

If we need to consider aligning with `TypeScript` (also the same as typescript-eslint), we should make it parse and then throw an error in Semantic instead.

Also, I found that the parameters printing is incorrect in codegen; the separator should be `,` not `|`.
https://github.com/oxc-project/oxc/blob/c1e6c9bfa2ff505dcc87c98495c7fb78185a217d/crates/oxc_codegen/src/gen.rs#L3490-L3496

Personally, I prefer to change the AST to only accept one parameter because multiple parameters means the code is incorrect and make it parse correctly is meaningless, the AST needs to deal with potential multiple parameters, moreover we can't assume only one parameter exists in codegen (ditto for formatted), we must iterate over `parameters` and print all of them, which increases unnecessary checks.

Yes, a lot of downside to parse it correctly.
